toil_orchestrator/tasks.py

```diff
@@ -205,6 +211,8 @@ def check_status_of_jobs(self):
                 except Exception as e:
                     error_message = "Failed to update job %s from file: %s\n%s" % (job.id, job_info_path,str(e))
                     logger.info(error_message)
+        elif not job.external_id and job.status == Status.PENDING:
+            continue
```

Comment:
Jobs in PENDING may or may not have an external ID. The working assumption with this PR is that once a Job is pending, submit_job_to_lsf will either successfully pull an external ID or fail. We a better way to handle this but for now this will work.